src/Transition_Systems-Weak_Transition_Systems.thy

```diff
@@ -36,6 +36,20 @@ notation weak.constant_bisimilarity (\<open>[\<approx>]\<close>)
 notation weak.simulation_up_to (\<open>weak'_sim\<^bsub>_\<^esub>\<close>)
 notation weak.bisimulation_up_to (\<open>weak'_bisim\<^bsub>_\<^esub>\<close>)
 
+lemma mutual_silent_weak_transitions_up_to_bisimilarity:
+  assumes "p \<Rightarrow>\<lparr>\<tau>\<rparr> p'" and "p' \<approx> q" and "q \<Rightarrow>\<lparr>\<tau>\<rparr> q'" and "q' \<approx> p"
+  shows "p \<approx> q"
+using assms
+proof (coinduction arbitrary: p q p' q' rule: weak.symmetric_up_to_rule [where \<F> = "[\<approx>]"])
+  case (simulation \<alpha> s)
+  from \<open>p \<Rightarrow>\<lparr>\<alpha>\<rparr> s\<close> and \<open>q \<Rightarrow>\<lparr>\<tau>\<rparr> q'\<close> and \<open>q' \<approx> p\<close> obtain s' where "q' \<Rightarrow>\<lparr>\<alpha>\<rparr> s'" and "s' \<approx> s"
+    by (blast elim: weak.bisimilarity.cases)
+  with \<open>q \<Rightarrow>\<lparr>\<tau>\<rparr> q'\<close> have "q \<Rightarrow>\<lparr>\<alpha>\<rparr> s'"
```

Comment:
Instead of `with ‹q ⇒⦇τ⦈ q'›`, you should say `from ‹q ⇒⦇τ⦈ q'› and ‹q' ⇒⦇α⦈ t›`, since `s ≈ t` isn’t needed here.

Reply:
> Instead of `with ‹q ⇒⦇τ⦈ q'›`, you should say `from ‹q ⇒⦇τ⦈ q'› and ‹q' ⇒⦇α⦈ t›`, since `s ≈ t` isn’t needed here.

Fixed in 4b231728b86e5c30dfffc003182b3a48a8db94ad, please verify.

Reply:
Looks good to me.